test/pytest/test_cnn_mnist.py

```diff
@@ -45,15 +45,19 @@ def keras_model(mnist_data):
   keras_model.fit(x_train, y_train, batch_size=32, epochs=5,  verbose=0)
   return keras_model
 
-@pytest.mark.parametrize('backend, io_type, strategy', [
+@pytest.mark.parametrize('backend,io_type,strategy', [
                                       ('Quartus', 'io_parallel', 'resource'),
+                                      ('Quartus', 'io_stream', 'resource'),
+  
                                       ('Vivado', 'io_parallel', 'resource'),
-                                      ('Vivado', 'io_parallel', 'latency')
+                                      ('Vivado', 'io_parallel', 'latency'),
+                                      ('Vivado', 'io_stream', 'latency'),
+                                      ('Vivado', 'io_stream', 'resource')
                                     ])
 def test_mnist_cnn(keras_model, mnist_data, backend, io_type, strategy):
   x_train, y_train, x_test, y_test = mnist_data
   
-  hls_config = hls4ml.utils.config_from_keras_model(keras_model, granularity='name')     
+  hls_config = hls4ml.utils.config_from_keras_model(keras_model, granularity='name', default_precision='ap_fixed<32, 9>')     
```

Comment:
That's odd, was it a one-off failed test?

Reply:
I ran all combinations twice, all io_parallel passed, all io_stream failed without this change.

Reply:
To summarize this issue, after a lot of debugging, turns out this change hides 4 separate bugs:
1. Vivado io_stream implementation of AveragePooling doesn't use a wider type to compute the sum needed in average like the io_parallel implementation does, hence the test fails with default settings. This will be addressed with a bit-exact flow in the future, but for now a simple tweak with `accum_t` would be enough for both io_stream implementations but...
2. Line-buffer implementation doesn't use `accum_t` for the sum, rather `data_T`.
3. In Quartus backend, after changes from #570 the `exp()` lookup table stores only negative values, but since the no saturation bits are used in io_parallel implementation of stable softmax strategy (that's a mouthful), we can pass the positive value (negative value wrapped around) and get incorrect results. This doesn't fail this test, but is incorrect nevertheless.
4. In io_stream implementation of stable softmax strategy, the saturation bits are used, so the most negative value (e.g, `0b1100000.0000` in binary) gets sliced to 0, and `exp(0) = 1`, so this throws off the result, hence the drop in accuracy and the test fails.

3 and 4. are addressed in this PR, I'll create a new one for 1. and 2. so as to not pollute this PR. Once that is in, we can remove this change and finally merge this PR.